src/main.rs

```diff
@@ -234,53 +266,55 @@ impl FuzzProject {
                     tsan_opts.push(':');
                 }
                 tsan_opts.push_str("report_signal_unsafe=0");
-                envs.push(("TSAN_OPTIONS", tsan_opts));
+                cmd.env("TSAN_OPTIONS", tsan_opts);
             }
 
             _ => {}
         }
 
-        cargo_args.push("--manifest-path".as_ref());
-        cargo_args.push(manifest_path.as_ref());
-        if release {
-            cargo_args.push("--release".as_ref());
-        }
-        cargo_args.push("--verbose".as_ref());
-        cargo_args.push("--bin".as_ref());
-        cargo_args.push(&target.as_ref());
-        //--target=<TARGET> won't pass rustflags to build scripts
-        cargo_args.push("--target".as_ref());
-        cargo_args.push(target_triple.as_ref());
+        Ok(cmd)
+    }
 
-        let mut cmd = process::Command::new("cargo");
-        cmd.arg("build")
-           .args(&cargo_args);
-        for &(ref k, ref v) in &envs { cmd.env(k, v); } // Command::envs still unstable
-        let status = cmd.status().chain_err(|| format!("could not execute: {:?}", cmd))?;
+    fn cmd(&self, args: &ArgMatches) -> Result<process::Command> {
+        let target = get_target(args)?;
+        let mut cmd = self.cargo("run", args)?;
+        let mut artefact_arg = ffi::OsString::from("-artifact_prefix=");
+        artefact_arg.push(self.artifacts_for(&target)?);
+
+        cmd.arg("--")
+           .arg(artefact_arg);
+
+        Ok(cmd)
+    }
+
+    /// Fuzz a given fuzz target
+    fn exec_fuzz<'a>(&self, args: &ArgMatches<'a>) -> Result<()> {
+        let target = get_target(args)?;
+
+        let mut cmd = self.cargo("build", args)?;
+        let status = cmd.status()
+            .chain_err(|| format!("could not execute: {:?}", cmd))?;
         if !status.success() {
             return Err(format!("could not build fuzz script: {:?}", cmd).into());
         }
 
-        let mut cmd = process::Command::new("cargo");
-        cmd.arg("run")
-           .args(&cargo_args);
-        for &(ref k, ref v) in &envs { cmd.env(k, v); } // Command::envs still unstable
+        let mut cmd = self.cmd(args)?;
```

Comment:
Yeah. There’s very little CLI flag reparsing going on, so it doesn’t matter much. Most of the work is already done by the time we get ArgMatches.